src/DocumentIndex.js

```diff
@@ -13,9 +13,9 @@ class DocumentIndex {
 
   updateIndex (oplog, onProgressCallback) {
     const reducer = (handled, item, idx) => {
-      if (item.payload.op === 'PUTALL') {
+      if (item.payload.op === 'PUTALL' && item.payload.docs[Symbol.iterator]) {
         for (const doc of item.payload.docs) {
-          if (handled[doc.key] !== true) {
+          if (doc && handled[doc.key] !== true) {
```

Comment:
Which case is this checking for, will one of the iterable results be null or undefined?

Reply:
Yes. as per @tabcat the assumption is that someone maliciously creates a payload that either has `doc` objects constructed in that fashion